compiler/codegen/OMRTreeEvaluator.cpp

```diff
@@ -147,6 +147,24 @@ bool OMR::TreeEvaluator::instanceOfOrCheckCastNeedSuperTest(TR::Node * node, TR:
    return false;
    }
 
+TR_GlobalRegisterNumber 
+OMR::TreeEvaluator::getHighGlobalRegisterNumberIfAny(TR::Node *node, TR::CodeGenerator *cg)
+   {
+    //No need for register pairs in 64-bit mode
+    if (TR::Compiler->target.is64Bit())
+        return -1;
+
+    //if the node itself doesn't have a type (e.g passthrough) we assume it has a child with a type
+    //However we keep track of the initial node as it will contain the register information.
+    TR::Node *rootNode = node;
+
+    while (node->getType() == TR::NoType)
+        node = node->getFirstChild();
+    TR_ASSERT(node && node->getType() != TR::NoType, "Expecting node %p, to have a specific type here", node);
```

Comment:
You're right, I've moved a separate assert into the while loop, this address the other concern as well, there should not be a NULL that is encountered when looking for a child with a Type